### PR TITLE
Update the retargeting application

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This PR:
- move the application files inside app/scripts
- update the application files
- update CMakeLists.txt to install the application files
- change the logic behind useVirtualizer inside OculusModule.cpp

__TODO__
- [x] test on the real robot
- [x] merge #12

### CODE REVIEW

**Comment by GiulioRomualdi** [comment]
Both the applications were tested today, and they are currently installed on the iCub laptop.  Waiting for #12.

File: modules/Oculus_module/src/OculusModule.cpp [hunk 1]
```diff
@@ -110,35 +110,33 @@ bool OculusModule::configureJoypad(const yarp::os::Searchable& config)
...
-        if (!YarpHelper::getDoubleFromSearchable(axisOptions, "deadzone", m_deadzone))
+        if (!YarpHelper::getDoubleFromSearchable(config, "deadzone", m_deadzone))
         {
             yError() << "[OculusModule::configureJoypad] Unable to find parameter deadzone";
             return false;
         }
-        if (!YarpHelper::getDoubleFromSearchable(axisOptions, "fullscale", m_fullscale))
+        if (!YarpHelper::getDoubleFromSearchable(config, "fullscale", m_fullscale))
         {
             yError() << "[OculusModule::configureJoypad] Unable to find parameter deadzone";
```
Comment (traversaro) [inline modules/Oculus_module/src/OculusModule.cpp]:
```suggestion
             yError() << "[OculusModule::configureJoypad] Unable to find parameter fullscale";
```

File: modules/Oculus_module/src/OculusModule.cpp [hunk 2]
```diff
@@ -110,35 +110,33 @@ bool OculusModule::configureJoypad(const yarp::os::Searchable& config)
...
             return false;
         }
-        if (!YarpHelper::getDoubleFromSearchable(axisOptions, "scale_X", m_scaleX))
+        if (!YarpHelper::getDoubleFromSearchable(config, "scale_X", m_scaleX))
         {
             yError() << "[OculusModule::configureJoypad] Unable to find parameter deadzone";
```
Comment (traversaro) [inline modules/Oculus_module/src/OculusModule.cpp]:
```suggestion
             yError() << "[OculusModule::configureJoypad] Unable to find parameter scale_X";
```

File: modules/Oculus_module/src/OculusModule.cpp [hunk 3]
```diff
@@ -110,35 +110,33 @@ bool OculusModule::configureJoypad(const yarp::os::Searchable& config)
...
             return false;
         }
-        if (!YarpHelper::getDoubleFromSearchable(axisOptions, "scale_Y", m_scaleY))
+        if (!YarpHelper::getDoubleFromSearchable(config, "scale_Y", m_scaleY))
         {
             yError() << "[OculusModule::configureJoypad] Unable to find parameter deadzone";
```
Comment (traversaro) [inline modules/Oculus_module/src/OculusModule.cpp]:
```suggestion
             yError() << "[OculusModule::configureJoypad] Unable to find parameter scale_Y";
```

File: app/robots/iCubGenova04/oculusConfig.ini [hunk 4]
```diff
@@ -21,19 +21,19 @@ robot                   icub
...
-# scale_Y                       5.0
-# use_lef                       1
-
 [OCULUS]
 root_frame_name                 mobile_base_body_link
 left_hand_frame_name            loculus
 right_hand_frame_name           roculus
 oculusOrientationPort           /oculusOrientation:i
 
+move_icub_using_joypad        1
```
Comment (traversaro) [inline app/robots/iCubGenova04/oculusConfig.ini]: In  `app/robots/iCubGenova04/oculusConfig.ini`, this options are under the `[JOYPAD]` group, while here no. It is normal?

Comment (GiulioRomualdi) [reply app/robots/iCubGenova04/oculusConfig.ini]: The idea was to remove the `[JOYPAD]` group and move the params, namely:
```
move_icub_using_joypad        1
deadzone                      0.3
fullscale                     1.0
scale_X                       5.0
scale_Y                       5.0
use_left                      1
```
into the `[OCULUS]` group

File: modules/Oculus_module/src/OculusModule.cpp (hunk 1 above)
Comment (GiulioRomualdi) [reply modules/Oculus_module/src/OculusModule.cpp]: Done in 160039a

File: modules/Oculus_module/src/OculusModule.cpp (hunk 2 above)
Comment (GiulioRomualdi) [reply modules/Oculus_module/src/OculusModule.cpp]: Done in 160039a

File: modules/Oculus_module/src/OculusModule.cpp (hunk 3 above)
Comment (GiulioRomualdi) [reply modules/Oculus_module/src/OculusModule.cpp]: Done in 160039a

File: app/robots/iCubGenova04/oculusConfig.ini (hunk 4 above)
Comment (GiulioRomualdi) [reply app/robots/iCubGenova04/oculusConfig.ini]: In 6d848a4 I fixed the `oculusConfig.ini` for icubGazeboSim

**Comment by traversaro** [comment]
`WIP` is just for testing on the robot or more code is expected to arrive in the PR?

**Comment by GiulioRomualdi** [comment]
`WIP` is because #12 has to be merged before this PR

**Pull request merged**